docs/en/interfaces/formats.md

```diff
@@ -476,6 +476,8 @@ The CSV format supports the output of totals and extremes the same way as `TabSe
 - [output_format_csv_crlf_end_of_line](/docs/en/operations/settings/settings-formats.md/#output_format_csv_crlf_end_of_line) - if it is set to true, end of line in CSV output format will be `\r\n` instead of `\n`. Default value - `false`.
 - [input_format_csv_skip_first_lines](/docs/en/operations/settings/settings-formats.md/#input_format_csv_skip_first_lines) - skip the specified number of lines at the beginning of data. Default value - `0`.
 - [input_format_csv_detect_header](/docs/en/operations/settings/settings-formats.md/#input_format_csv_detect_header) - automatically detect header with names and types in CSV format. Default value - `true`.
+- [input_format_csv_with_names](/docs/en/operations/settings/settings-formats.md/#input_format_csv_with_names) - get names from data file in CSV format. Default value - `false`.
+- [input_format_csv_with_types](/docs/en/operations/settings/settings-formats.md/#input_format_csv_with_types) - get types from data file in CSV format. Default value - `false`.
```

Comment:
Is it applied only to `HiveText` format? Maybe alternatively we can introduce `HiveTextWithNamesAndTypes` instead of adding settings. Or at least names should refer to `hive`

Reply:
We definitely should not add such settings and introduce `HiveTextWithNames` and `HiveTextWithNamesAndTypes` formats. And I see that by default `HiveText` uses `with_names=true`, so it can be backward-incompatible change. But in CSV when `with_names=false` and `with_types=false` we try to detect header automatically, so should not be a problem to make `HiveText` without names and types by default and add  `HiveTextWithNames` and `HiveTextWithNamesAndTypes` formats.

Reply:
Sorry, this pr is not necessary